File: java/07 - Demo 3 - Refactoring/src/main/java/replacetempwithquery/ProductPrice.java
package replacetempwithquery;

public class ProductPrice {

	private double quantity;
	private double itemPrice;

	public ProductPrice(double quantity, double itemPrice) {
		super();
		this.quantity = quantity;
		this.itemPrice = itemPrice;
	}

	double getPrice() {
		double basePrice = quantity * itemPrice;
		double discountFactor;
		if (basePrice > 1000)
			discountFactor = 0.95;
		else
			discountFactor = 0.98;
		return basePrice * discountFactor;
	}

}

File: csharp/07 - Demo 3 - Refactoring/Refactoring/ReplaceTempWithQuery/ProductPrice.cs
namespace Refactoring.ReplaceTempWithQuery
{
    public class ProductPrice
    {
        private double _quantity;
        private double _itemPrice;

        public ProductPrice(double quantity, double itemPrice)
        {
            _quantity = quantity;
            _itemPrice = itemPrice;
        }

        public double Price
        {
            get
            {
                double basePrice = _quantity * _itemPrice;
                double discountFactor;
                if (basePrice > 1000)
                    discountFactor = 0.95;
                else
                    discountFactor = 0.98;
                return basePrice * discountFactor;
            }
        }

    }
}